sns_ik_lib/utilities/sns_linear_solver.cpp

```diff
@@ -0,0 +1,85 @@
+/** @file sns_linear_solver.cpp
+ *
+ * @brief Linear Solver, used by the SNS-IK velocity IK solvers.
+ * @author: Matthew Kelly
+ *
+ * This class is used to solve linear systems. It is a wrapper for two different internal solvers:
+ *   --> psuedo-inverse solver: included for legacy support on Eigen 3.2.0
+ *   --> direct linear solver: prefered solver when available. Requries Eigen 3.3.4
+ *
+ *    Copyright 2018 Rethink Robotics
```

Comment:
The copyright should be first before the description of the file, and should have its own set of comment braces for clarity

Reply:
updated!